src/ert/_c_wrappers/enkf/observations/obs_vector.py

```diff
@@ -16,89 +13,58 @@
-    _get_impl_type = ResPrototype(
-        "enkf_obs_impl_type obs_vector_get_impl_type( obs_vector)"
-    )
-    _install_node = ResPrototype(
-        "void  obs_vector_install_node(obs_vector, int, void*)"
-    )
-    _get_next_active_step = ResPrototype(
-        "int   obs_vector_get_next_active_step(obs_vector, int)"
-    )
-    _get_obs_key = ResPrototype("char*  obs_vector_get_obs_key(obs_vector)")
-
+class ObsVector:
     def __init__(
         self,
         observation_type: EnkfObservationImplementationType,
         observation_key: str,
         config_node_key: str,
-        num_reports: int,
+        num_steps: int,
     ):
-        assert isinstance(observation_type, EnkfObservationImplementationType)
-        assert isinstance(observation_key, str)
-        assert isinstance(config_node_key, str)
-        assert isinstance(num_reports, int)
-        c_ptr = self._alloc(
-            observation_type, observation_key, config_node_key, num_reports
-        )
-        super().__init__(c_ptr)
+        self.observation_type = observation_type
+        self.observation_key = observation_key
+        self.config_node_key = config_node_key
+        self.nodes: List[Optional[Union[SummaryObservation, GenObservation]]] = [
+            None
+        ] * num_steps
+        self.steps = set()
 
     def getDataKey(self) -> str:
-        return self._get_state_kw()
+        return self.config_node_key
 
     def getObservationKey(self) -> str:
-        return self.getKey()
+        return self.observation_key
 
     def getKey(self) -> str:
-        return self._get_key()
+        return self.observation_key
 
     def getObsKey(self) -> str:
-        return self._get_obs_key()
+        return self.observation_key
 
-    def getNode(self, index: int) -> Union[SummaryObservation, GenObservation]:
-        pointer = self._iget_node(index)
-
-        node_type = self.getImplementationType()
-        if node_type == EnkfObservationImplementationType.SUMMARY_OBS:
-            return SummaryObservation.createCReference(pointer, self)
-        elif node_type == EnkfObservationImplementationType.GEN_OBS:
-            return GenObservation.createCReference(pointer, self)
-        else:
-            raise AssertionError(f"Node type '{node_type}' currently not supported!")
+    def getNode(
+        self, index: int
+    ) -> Optional[Union[SummaryObservation, GenObservation]]:
+        return self.nodes[index]
 
-    def __iter__(self):
+    def __iter__(self) -> Iterable[Union[SummaryObservation, GenObservation]]:
         """Iterate over active report steps; return node"""
-        for step in self.getStepList():
-            yield self.getNode(step)
+        return (self.nodes[i] for i in self.getStepList())  # type:ignore
 
     def getStepList(self) -> List[int]:
         """
         Will return an IntVector with the active report steps.
         """
-        return _clib.obs_vector_get_step_list(self)
+        return sorted(self.steps)
```

Comment:
Is it better to sort on insert? I assume this happens more often than insertion?

Reply:
Changed it to a `SortedList`